<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>ThreadPoolShell</class>
 <widget class="QMainWindow" name="ThreadPoolShell">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>829</width>
    <height>794</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Advanced Thread Pool</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <layout class="QGridLayout" name="gridLayout_4">
    <item row="0" column="0">
     <layout class="QVBoxLayout" name="verticalLayout_5">
      <item>
       <layout class="QHBoxLayout" name="horizontalLayout_2">
        <item>
         <widget class="QLabel" name="label">
          <property name="text">
           <string>Unshared threads quantity:</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QLineEdit" name="lineEdit_unshared_threads">
          <property name="sizePolicy">
           <sizepolicy hsizetype="Fixed" vsizetype="Fixed">
            <horstretch>0</horstretch>
            <verstretch>0</verstretch>
           </sizepolicy>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QLabel" name="label_13">
          <property name="text">
           <string>Shared threads quantity:</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QLineEdit" name="lineEdit_shared_threads">
          <property name="sizePolicy">
           <sizepolicy hsizetype="Fixed" vsizetype="Fixed">
            <horstretch>0</horstretch>
            <verstretch>0</verstretch>
           </sizepolicy>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QComboBox" name="comboBox_Affinity">
          <item>
           <property name="text">
            <string>No affinity</string>
           </property>
          </item>
          <item>
           <property name="text">
            <string>Yes affinity</string>
           </property>
          </item>
         </widget>
        </item>
        <item>
         <spacer name="horizontalSpacer_2">
          <property name="orientation">
           <enum>Qt::Horizontal</enum>
          </property>
          <property name="sizeHint" stdset="0">
           <size>
            <width>40</width>
            <height>20</height>
           </size>
          </property>
         </spacer>
        </item>
        <item>
         <widget class="QLabel" name="label_2">
          <property name="text">
           <string>STATE:</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QLabel" name="label_state">
          <property name="frameShape">
           <enum>QFrame::Box</enum>
          </property>
          <property name="frameShadow">
           <enum>QFrame::Raised</enum>
          </property>
          <property name="text">
           <string>POWER ON</string>
          </property>
         </widget>
        </item>
       </layout>
      </item>
      <item>
       <widget class="QTabWidget" name="tabWidget">
        <property name="currentIndex">
         <number>1</number>
        </property>
        <widget class="QWidget" name="tab">
         <attribute name="title">
          <string>Unshared threads (long tasks)</string>
         </attribute>
         <layout class="QGridLayout" name="gridLayout_2">
          <item row="0" column="0">
           <widget class="QTreeWidget" name="treeWidget_notshared_threads">
            <property name="columnCount">
             <number>4</number>
            </property>
            <column>
             <property name="text">
              <string>Thread ID</string>
             </property>
            </column>
            <column>
             <property name="text">
              <string>Type</string>
             </property>
            </column>
            <column>
             <property name="text">
              <string>Task description</string>
             </property>
            </column>
            <column>
             <property name="text">
              <string>Affinity mask</string>
             </property>
            </column>
           </widget>
          </item>
         </layout>
        </widget>
        <widget class="QWidget" name="tab_2">
         <attribute name="title">
          <string>Shared threads (short and repeat tasks)</string>
         </attribute>
         <layout class="QGridLayout" name="gridLayout_3">
          <item row="0" column="0">
           <widget class="QTreeWidget" name="treeWidget_shared_threads">
            <column>
             <property name="text">
              <string>Thread ID</string>
             </property>
            </column>
            <column>
             <property name="text">
              <string>Average task quantity</string>
             </property>
            </column>
            <column>
             <property name="text">
              <string>Affinity mask</string>
             </property>
            </column>
           </widget>
          </item>
         </layout>
        </widget>
       </widget>
      </item>
      <item>
       <layout class="QHBoxLayout" name="horizontalLayout_4">
        <item>
         <layout class="QVBoxLayout" name="verticalLayout_3">
          <item>
           <widget class="QLabel" name="label_14">
            <property name="text">
             <string>Repeat tasks info:</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QTreeWidget" name="treeWidget_repeat_tasks">
            <column>
             <property name="text">
              <string>Task ID</string>
             </property>
            </column>
            <column>
             <property name="text">
              <string>Time</string>
             </property>
            </column>
            <column>
             <property name="text">
              <string>Who</string>
             </property>
            </column>
           </widget>
          </item>
         </layout>
        </item>
        <item>
         <layout class="QVBoxLayout" name="verticalLayout_4">
          <item>
           <widget class="QGroupBox" name="groupBox">
            <property name="title">
             <string>Thread pool information:</string>
            </property>
            <layout class="QGridLayout" name="gridLayout">
             <item row="0" column="0">
              <layout class="QHBoxLayout" name="horizontalLayout_3">
               <item>
                <layout class="QVBoxLayout" name="verticalLayout">
                 <item>
                  <widget class="QLabel" name="label_17">
                   <property name="text">
                    <string>Core quantity:</string>
                   </property>
                  </widget>
                 </item>
                 <item>
                  <widget class="QLabel" name="label_3">
                   <property name="text">
                    <string>Unshared threads quantity:</string>
                   </property>
                  </widget>
                 </item>
                 <item>
                  <widget class="QLabel" name="label_5">
                   <property name="text">
                    <string>Shared threads quantity:</string>
                   </property>
                  </widget>
                 </item>
                 <item>
                  <widget class="QLabel" name="label_7">
                   <property name="text">
                    <string>Long tasks quantity:</string>
                   </property>
                  </widget>
                 </item>
                 <item>
                  <widget class="QLabel" name="label_9">
                   <property name="text">
                    <string>Repeat tasks quantity:</string>
                   </property>
                  </widget>
                 </item>
                </layout>
               </item>
               <item>
                <layout class="QVBoxLayout" name="verticalLayout_2">
                 <item>
                  <widget class="QLabel" name="label_CoreQuantity">
                   <property name="frameShape">
                    <enum>QFrame::Panel</enum>
                   </property>
                   <property name="frameShadow">
                    <enum>QFrame::Sunken</enum>
                   </property>
                   <property name="text">
                    <string>4</string>
                   </property>
                  </widget>
                 </item>
                 <item>
                  <widget class="QLabel" name="label_UnsharedQ">
                   <property name="frameShape">
                    <enum>QFrame::Panel</enum>
                   </property>
                   <property name="frameShadow">
                    <enum>QFrame::Sunken</enum>
                   </property>
                   <property name="text">
                    <string>10</string>
                   </property>
                  </widget>
                 </item>
                 <item>
                  <widget class="QLabel" name="label_SharedQ">
                   <property name="frameShape">
                    <enum>QFrame::Panel</enum>
                   </property>
                   <property name="frameShadow">
                    <enum>QFrame::Sunken</enum>
                   </property>
                   <property name="text">
                    <string>10</string>
                   </property>
                  </widget>
                 </item>
                 <item>
                  <widget class="QLabel" name="label_longTaskQ">
                   <property name="frameShape">
                    <enum>QFrame::Panel</enum>
                   </property>
                   <property name="frameShadow">
                    <enum>QFrame::Sunken</enum>
                   </property>
                   <property name="text">
                    <string>0</string>
                   </property>
                  </widget>
                 </item>
                 <item>
                  <widget class="QLabel" name="label_repeatTaskQ">
                   <property name="frameShape">
                    <enum>QFrame::Panel</enum>
                   </property>
                   <property name="frameShadow">
                    <enum>QFrame::Sunken</enum>
                   </property>
                   <property name="text">
                    <string>0</string>
                   </property>
                  </widget>
                 </item>
                </layout>
               </item>
              </layout>
             </item>
            </layout>
           </widget>
          </item>
          <item>
           <spacer name="verticalSpacer">
            <property name="orientation">
             <enum>Qt::Vertical</enum>
            </property>
            <property name="sizeHint" stdset="0">
             <size>
              <width>20</width>
              <height>40</height>
             </size>
            </property>
           </spacer>
          </item>
         </layout>
        </item>
       </layout>
      </item>
     </layout>
    </item>
    <item row="1" column="0">
     <layout class="QHBoxLayout" name="horizontalLayout">
      <item>
       <widget class="QPushButton" name="pushButton_StopPool">
        <property name="text">
         <string>Stop pool</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QPushButton" name="pushButton_StartPool">
        <property name="text">
         <string>Start pool</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QPushButton" name="pushButton_Apply">
        <property name="text">
         <string>Apply</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QPushButton" name="pushButton_Close">
        <property name="text">
         <string>Close</string>
        </property>
       </widget>
      </item>
      <item>
       <spacer name="horizontalSpacer">
        <property name="orientation">
         <enum>Qt::Horizontal</enum>
        </property>
        <property name="sizeHint" stdset="0">
         <size>
          <width>40</width>
          <height>20</height>
         </size>
        </property>
       </spacer>
      </item>
     </layout>
    </item>
   </layout>
  </widget>
  <widget class="QMenuBar" name="menubar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>829</width>
     <height>22</height>
    </rect>
   </property>
  </widget>
  <widget class="QStatusBar" name="statusbar"/>
 </widget>
 <resources/>
 <connections/>
</ui>
